Null pointer exception on ValueMap

Hello,

We're getting a lot of NullPointerException when calling identify of Analytics. The version we are using is the **4.1.6**

And the point where we get it:
```
Caused by: java.lang.NullPointerException
        at java.util.concurrent.ConcurrentHashMap.putVal(ConcurrentHashMap.java:821)
        at java.util.concurrent.ConcurrentHashMap.putAll(ConcurrentHashMap.java:894)
        at com.segment.analytics.ValueMap.putAll(ValueMap.java:99)
        at com.segment.analytics.Analytics.identify(Analytics.java:400)
```

Did you already got some issue like this before?

Thanks!

--- a/analytics-tests/src/test/java/com/segment/analytics/internal/UtilsTest.java
+++ b/analytics-tests/src/test/java/com/segment/analytics/internal/UtilsTest.java
@@ -105,4 +105,24 @@ public class UtilsTest {
     when(context.checkCallingOrSelfPermission(ACCESS_NETWORK_STATE)).thenReturn(PERMISSION_DENIED);
     assertThat(isConnected(context)).isTrue();
   }
+
+  @Test public void nullableConcurrentHashMapPutIgnoresNulls() throws Exception {
+    Map<String, String> map = new Utils.NullableConcurrentHashMap<>();
+    map.put(null, null);
+    map.put("foo", null);
+    map.put(null, "bar");
+
+    assertThat(map).isEmpty();
+  }
+
+  @Test public void nullableConcurrentHashMapPutAllIgnoresNulls() throws Exception {
+    Map<String, String> values = new LinkedHashMap<>();
+    values.put(null, null);
+    values.put("foo", null);
+    values.put(null, "bar");
+
+    Map<String, String> map = new Utils.NullableConcurrentHashMap<>();
+    map.putAll(values);
+    assertThat(map).isEmpty();
+  }
 }

--- a/analytics/src/main/java/com/segment/analytics/internal/Utils.java
+++ b/analytics/src/main/java/com/segment/analytics/internal/Utils.java
@@ -426,5 +426,11 @@ public final class Utils {
       }
       return super.put(key, value);
     }
+
+    @Override public void putAll(Map<? extends K, ? extends V> m) {
+      for (Map.Entry<? extends K, ? extends V> e : m.entrySet()) {
+        put(e.getKey(), e.getValue());
+      }
+    }
   }
 }

